# Introduction

After watching *The Last Dance* and seeing Jordan and the Bulls go into the playoffs with the highest regular season win-loss record of any team up until then, I got curious about which team has gone into the playoffs with the worst win-loss record and came out finals champions.

## Dataset

So I need to compile a dataset which has the season win-loss record of each team and the results of the Finals. Luckily I came across a Kaggle dataset which already compiled that from basketball-reference. I'll do some eyeball verification after I compile a list of the Finals champions and their season win-loss record from the dataset. It also only has records up until the 2016-17 season so I'll have to supplement the last few seasons with my own research.

In [19]:
import pandas as pd
import numpy as np
from matplotlib import pyplot

In [6]:
df = pd.read_csv('data/Team_Records.csv')
print(df.head())

    Season   Lg             Team   W   L   W/L%  Finish   SRS  Pace  Rel_Pace  \
0  2017-18  NBA   Boston Celtics  29  10  0.744       1  4.38  95.4      -1.7   
1  2016-17  NBA  Boston Celtics*  53  29  0.646       1  2.25  96.8       0.4   
2  2015-16  NBA  Boston Celtics*  48  34  0.585       2  2.84  98.5       2.7   
3  2014-15  NBA  Boston Celtics*  40  42  0.488       2 -0.40  95.8       1.9   
4  2013-14  NBA   Boston Celtics  25  57  0.305       4 -4.97  93.3      -0.6   

    ORtg  Rel_ORtg   DRtg  Rel_DRtg                Playoffs  \
0  108.0       0.2  102.8      -5.0                     NaN   
1  111.2       2.4  108.4      -0.4    Lost E. Conf. Finals   
2  106.8       0.4  103.6      -2.8  Lost E. Conf. 1st Rnd.   
3  104.7      -0.9  104.5      -1.1  Lost E. Conf. 1st Rnd.   
4  102.9      -3.8  107.7       1.0                     NaN   

              Coaches            Top WS  
0  B. Stevens (29-10)   K. Irving (5.7)  
1  B. Stevens (53-29)  I. Thomas (12.5)  
2  B. St

In [7]:
df.Playoffs.unique()

array([nan, 'Lost E. Conf. Finals', 'Lost E. Conf. 1st Rnd.',
       'Lost E. Conf. Semis', 'Lost Finals', 'Won Finals',
       'Lost E. Div. Finals', 'Lost E. Div. Semis', 'Lost Quarterfinals',
       'Lost E. Div. Third Place Tiebreaker',
       'Eliminated in E. Div. Rnd. Robin', 'Lost Semis',
       'Lost W. Conf. Finals', 'Lost W. Conf. 1st Rnd.',
       'Lost W. Conf. Semis', 'Lost W. Div. Finals',
       'Lost W. Div. Tiebreaker', 'Lost W. Div. Semis',
       'Eliminated in W. Div. Rnd. Robin', 'Lost Central Div. Finals',
       'Lost 1st Rnd.', 'Lost Central Div. Semis'], dtype=object)

In [71]:
winners = df.loc[df['Playoffs'] == 'Won Finals']
print(winners.loc[winners['Season'] == "2016-17"]) # Print the last season record available

def add_row_to_df(df, szn, team, w, l, wl_pct):
    entry = {'Season': szn, 'Lg': 'NBA', 'Team': team, 'W': w, 'L': l, 'W/L%': wl_pct, 'Playoffs': 'Won Finals'}
    df = df.append(entry, ignore_index=True)
    return df

winners = add_row_to_df(winners, '2017-18', 'Golden State Warriors', 58, 24, .707)
winners = add_row_to_df(winners, '2018-19', 'Toronto Raptors', 58, 24, .707)
winners = add_row_to_df(winners, '2019-20', 'Los Angeles Lakers', 52, 19, .732)

print(winners.loc[winners['Season'] == "2019-20"]) # Print the last season record available

      Season   Lg                    Team   W   L   W/L%  Finish    SRS  Pace  \
782  2016-17  NBA  Golden State Warriors*  67  15  0.817       1  11.35  99.8   

     Rel_Pace   ORtg  Rel_ORtg   DRtg  Rel_DRtg    Playoffs          Coaches  \
782       3.4  115.6       6.8  104.0      -4.8  Won Finals  S. Kerr (67-15)   

              Top WS  
782  S. Curry (12.6)  
     Season   Lg                Team   W   L   W/L%  Finish  SRS  Pace  \
77  2019-20  NBA  Los Angeles Lakers  52  19  0.732     NaN  NaN   NaN   

    Rel_Pace  ORtg  Rel_ORtg  DRtg  Rel_DRtg    Playoffs Coaches Top WS  
77       NaN   NaN       NaN   NaN       NaN  Won Finals     NaN    NaN  


In [66]:
print(winners.sort_values(by="W/L%"))

     Season   Lg                    Team   W   L   W/L%  Finish    SRS   Pace  \
62  1977-78  NBA     Washington Bullets*  44  38  0.537     2.0   0.82  108.1   
32  1971-72  ABA         Indiana Pacers*  47  37  0.560     2.0   2.72  110.7   
66  1957-58  NBA        St. Louis Hawks*  41  31  0.569     1.0   0.82  119.5   
67  1994-95  NBA        Houston Rockets*  47  35  0.573     3.0   2.32   94.2   
44  1946-47  BAA  Philadelphia Warriors*  35  25  0.583     2.0   3.16    NaN   
..      ...  ...                     ...  ..  ..    ...     ...    ...    ...   
38  1991-92  NBA          Chicago Bulls*  67  15  0.817     1.0  10.07   94.4   
20  1966-67  NBA     Philadelphia 76ers*  68  13  0.840     1.0   8.50  122.9   
56  1971-72  NBA     Los Angeles Lakers*  69  13  0.841     1.0  11.65  116.9   
35  1996-97  NBA          Chicago Bulls*  69  13  0.841     1.0  10.70   90.0   
36  1995-96  NBA          Chicago Bulls*  72  10  0.878     1.0  11.80   91.1   

    Rel_Pace   ORtg  Rel_OR

So the Washington Bullets (aka the Wizards) are the NBA champions with the lowest ever season win-loss percentage of 53.7%. Interestingly enough, the highest regular season record is held by the 2015-16 Warriors, at 73-9 (89%), but since they lost to the Cavs in the finals they do not appear on this list. The Finals champions with the best regular season record are still the '96 Bulls.